File: app/src/main/java/lemond/annoying/gamerscompanion/repository/objects/Image.java
package lemond.annoying.gamerscompanion.repository.objects;


import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

public class Image implements Parcelable {
    @SerializedName("url")
    public String url;
    @SerializedName("cloudinary_id")
    public String cloudinary_id;

    protected Image(Parcel in) {
        url = in.readString();
        cloudinary_id = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(url);
        dest.writeString(cloudinary_id);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Image> CREATOR = new Creator<Image>() {
        @Override
        public Image createFromParcel(Parcel in) {
            return new Image(in);
        }

        @Override
        public Image[] newArray(int size) {
            return new Image[size];
        }
    };
}

File: app/src/main/java/lemond/annoying/gamerscompanion/repository/objects/Image.kt
package lemond.annoying.gamerscompanion.repository.objects


import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.SerializedName

class Image protected constructor(parcel: Parcel) : Parcelable {
    @SerializedName("url")
    var url: String = parcel.readString()!!
    @SerializedName("cloudinary_id")
    var cloudinary_id: String = parcel.readString()!!

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(url)
        dest.writeString(cloudinary_id)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        @JvmField
        val CREATOR = object : Parcelable.Creator<Image> {
            override fun createFromParcel(parcel: Parcel): Image {
                return Image(parcel)
            }

            override fun newArray(size: Int): Array<Image?> {
                return arrayOfNulls(size)
            }
        }
    }
}
